src/wp-includes/rest-api/endpoints/class-wp-rest-pattern-directory-controller.php

```diff
@@ -144,6 +189,8 @@ public function get_items( $request ) {
 			}
 		}
 
+		set_site_transient( $transient_key, $response, HOUR_IN_SECONDS );
```

Comment:
The caching strategy doesn't seem right to me since there's a filter in `prepare_item_for_response`. I think the cache should only include `$raw_patterns`. Otherwise you'd have to wait for an hour until a newly added filter will work?

Reply:
Ah, that's a great point, I'll test that out and adjust.

Reply:
Yep, you were right. That's a great catch, thanks!

Fixed in cce106fca5fbb75808337604c9e20ea21d273dad

I also made a few tweaks to minimize diff noise against `master`, so viewing that commit w/ whitespace ignored might be easier, or viewing the entire branch diff